core-data/test/resolvers.js

```diff
@@ -30,3 +30,21 @@ describe( 'getCategories', () => {
 		expect( received ).toEqual( receiveTerms( 'categories', CATEGORIES ) );
 	} );
 } );
+
+describe( 'getMedia', () => {
+	const MEDIA = { id: 1 };
+
+	beforeAll( () => {
+		apiRequest.mockImplementation( ( options ) => {
+			if ( options.path === '/wp/v2/media/1' ) {
+				return Promise.resolve( MEDIA );
+			}
+		} );
+	} );
+
+	it( 'yields with requested media', async () => {
```

Comment:
I really like these tests. Nice work setting up the pattern @aduth 

It also shows that we could push it further by avoiding the mocks entirely (using `call` effect) but granted that it's not that important at the moment.